include/swift/AST/Decl.h

```diff
@@ -1864,76 +1864,104 @@ class ExtensionRange {
 /// Pattern and Initialization expression.  The pattern is always present, but
 /// the initializer can be null if there is none.
 class PatternBindingEntry {
-  Pattern *ThePattern;
-
-  /// The location of the equal '=' token.
-  SourceLoc EqualLoc;
-
   enum class Flags {
     Checked = 1 << 0,
     Removed = 1 << 1,
-    Lazy    = 1 << 2,
+    Lazy    = 1 << 2
   };
+  llvm::PointerIntPair<Pattern *, 3, OptionSet<Flags>> PatternAndFlags;
+
+  /// The location of the equal '=' token.
+  SourceLoc EqualLoc;
```

Comment:
Do you think we should put this into the union too? It appeals to my "pack everything as small as possible" instincts, but I can also see us doing a better job preserving source locations through deserialization at some point.

Reply:
Do we serialize any SourceLocs currently? I feel like we could get the win here by tying the EqualLoc with the InitExpr, and then pay for it later if we do end up serializing this loc. I don't think it would hurt to throw it in the union now.

Reply:
We don't today, on the grounds that the serialized module might be distributed somewhere. Okay then.